BigQuery: systest 'test_lists_datasets_by_label' failure

See: https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/7193

```python
_________________________ test_list_datasets_by_label __________________________

client = <google.cloud.bigquery.client.Client object at 0x7f6cd03273d0>
to_delete = [Dataset(DatasetReference(u'precise-truck-742', u'list_datasets_by_label_1531507522596'))]

    def test_list_datasets_by_label(client, to_delete):
        dataset_id = 'list_datasets_by_label_{}'.format(_millis())
        dataset = bigquery.Dataset(client.dataset(dataset_id))
        dataset.labels = {'color': 'green'}
        dataset = client.create_dataset(dataset)  # API request
        to_delete.append(dataset)
    
        # [START bigquery_list_datasets_by_label]
        # from google.cloud import bigquery
        # client = bigquery.Client()
    
        # The following label filter example will find datasets with an
        # arbitrary 'color' label set to 'green'
        label_filter = 'labels.color:green'
        datasets = list(client.list_datasets(filter=label_filter))
    
        if datasets:
            print('Datasets filtered by {}:'.format(label_filter))
            for dataset in datasets:  # API request(s)
                print('\t{}'.format(dataset.dataset_id))
        else:
            print('No datasets found with this filter.')
        # [END bigquery_list_datasets_by_label]
>       assert len(datasets) == 1
E       assert 2 == 1
E        +  where 2 = len([<google.cloud.bigquery.dataset.DatasetListItem object at 0x7f6cd03a5e50>, <google.cloud.bigquery.dataset.DatasetListItem object at 0x7f6cd040fd90>])

../docs/bigquery/snippets.py:173: AssertionError
```

Likely overlapping test runs.  Suggested fix is to use a label value with `_unique_id`, or else weaken the assertion (assert that the current dataset ID is in the set returned by the `list_datasets` call).

--- a/docs/bigquery/snippets.py
+++ b/docs/bigquery/snippets.py
@@ -168,7 +168,8 @@ def test_list_datasets_by_label(client, to_delete):
     else:
         print('No datasets found with this filter.')
     # [END bigquery_list_datasets_by_label]
-    assert len(datasets) == 1
+    found = set([dataset.dataset_id for dataset in datasets])
+    assert dataset_id in found
 
 
 def test_create_dataset(client, to_delete):

